test/new-e2e/tests/agent-platform/install-script/install_script_test.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+package installscript
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common"
+	filemanager "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/file-manager"
+	helpers "github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/common/helper"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/agent-platform/install"
+	e2eOs "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+
+	"testing"
+)
+
+var osVersion = flag.String("osversion", "", "os version to test")
+var platform = flag.String("platform", "", "platform to test")
+var cwsSupportedOsVersion = flag.String("cws-supported-osversion", "", "list of os where CWS is supported")
+
+type installScriptSuite struct {
+	e2e.Suite[e2e.VMEnv]
+	cwsSupported bool
+}
+
+func TestInstallScript(t *testing.T) {
+	platformJSON := map[string]map[string]string{}
+	platformFileContent, err := os.ReadFile("../platforms.json")
+	if err != nil {
+		panic(fmt.Sprintf("failed to read platform file: %v", err))
+	}
+	err = json.Unmarshal(platformFileContent, &platformJSON)
+	if err != nil {
+		panic(fmt.Sprintf("failed to umarshall platform file: %v", err))
+	}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
You can use a `require` that panics on assertion failure
```suggestion
        require.NoError(t, err, fmt.Sprintf("failed to unmarshall platform file: %v", err))
```